File: LeetCode/26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.go
package main

import (
	"fmt"
	"reflect"
)

func removeDuplicates(nums []int) int {
	current := nums[0]
	i := 1
	j := 1
	for i < len(nums) {
		if current != nums[i] {
			nums[j] = nums[i]
			j += 1
			current = nums[i]
		}
		i += 1
	}
	return j
}

type Quest struct {
	inp []int
	sol []int
	pos int
}

func checkAns(q Quest) {
	out := removeDuplicates(q.inp)
	if !reflect.DeepEqual(q.sol, q.inp[:out]) {
		fmt.Printf("%v != %v", q.inp, q.sol)
	}
}

func main() {
	var q Quest

	q = Quest{[]int{1, 1, 2}, []int{1, 2}, 2}
	checkAns(q)

	q = Quest{[]int{0, 0, 1, 1, 1, 2, 2, 3, 3, 4}, []int{0, 1, 2, 3, 4}, 5}
	checkAns(q)
}

File: LeetCode/26-remove-duplicates-from-sorted-array/26-remove-duplicates-from-sorted-array.c
#include <stdio.h>

int removeDuplicates(int *nums, int numsSize)
{
    int current = nums[0];
    int i = 1, j = 1;
    while (i < numsSize)
    {
        if (current != nums[i])
        {
            nums[j] = nums[i];
            j += 1;
            current = nums[i];
        }
        i += 1;
    }
    return j;
}

struct quest
{
    int *inp;
    int size;
    int *sol;
    int out;
};
typedef struct quest Quest;

int checkAns(Quest q)
{
    int out = removeDuplicates(q.inp, q.size);

    if (out != q.out)
    {
        printf("out != q.out, %d != %d\n", out, q.out);
        return 1;
    }

    for (int i = 0; i < out; i++)
    {

        if (q.inp[i] != q.sol[i])
        {
            printf("q.inp != q.sol, %d != %d\n", q.inp[i], q.sol[i]);
            return 0;
        }
    }
    return 1;
}

int main()
{
    Quest q;
    q = (Quest){(int[]){1, 1, 2}, 3, (int[]){1, 2}, 2};
    checkAns(q);

    return 0;
}